Update EIP-8032: Transition changes and add clarity (#10619)
ethereum/EIPs: EIPS/eip-8032.md

   - Increment `cursor_accum` slot for every slot hash seen
   - Stop when either:
     - `TRANSITION_SLOTS_PER_BLOCK` slots have been processed, or
     - there are no more storage slots for the account.

3. **Finalize account (if exhausted)**  
   If all storage slots for `cursor_account_hash` have been visited, set `storage_count(cursor_account)` to `cursor_accum` in the account object.
   
   If `TRANSITION_MAX_ACCOUNTS` accounts are iterated, then stop iterating.
   - Else: 
        - Advance to the next account:  
        `cursor_account_hash = next account hash that is present in the state trie`  
        - Reset per-account cursors:  
        `cursor_slot_hash = 0x00…00`, `cursor_accum = 0`

4. **Persist**  
   If either `TRANSITION_SLOTS_PER_BLOCK` or `TRANSITION_MAX_ACCOUNTS` is reached, write `cursor_account_hash`, `cursor_slot_hash`, and `cursor_accum` to the registry in the post-state of the block.

5. **Completion**  

>    The transition is complete when there is no account present in the state trie whose account hash is lexicographically greater than the current `cursor_account_hash`.

Reviewer: ```suggestion
   The transition is complete when there is no further account present in the trie, with an account hash that is lexicographically greater than the current `cursor_account_hash`.
```
